src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -14225,89 +14225,1275 @@ public static void Main()
-                Diagnostic(ErrorCode.ERR_NotNullRefDefaultParameter, "f").WithArguments("f", "object").WithLocation(1, 20),
-                // (1,24): error CS0518: Predefined type 'System.Double' is not defined or imported
-                // var lam1 = (object f = 1.0) => f;
-                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "1.0").WithArguments("System.Double").WithLocation(1, 24),
-                // (2,20): error CS1763: 'd' is of type 'object'. A default parameter value of a reference type other than string can only be initialized with null
-                // var lam2 = (object d = 2m) => d;
-                Diagnostic(ErrorCode.ERR_NotNullRefDefaultParameter, "d").WithArguments("d", "object").WithLocation(2, 20),
-                // (2,24): error CS0518: Predefined type 'System.Decimal' is not defined or imported
-                // var lam2 = (object d = 2m) => d;
-                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "2m").WithArguments("System.Decimal").WithLocation(2, 24));
+            var verifier = CompileAndVerify(source, expectedOutput: """
+                <>f__AnonymousDelegate0`1[System.Decimal]
+                <>f__AnonymousDelegate0`1[System.Decimal]
+                <>f__AnonymousDelegate0`1[System.Decimal]
+                System.Action`1[System.Decimal]
+                <>f__AnonymousDelegate0`1[System.Nullable`1[System.Decimal]]
+                <>f__AnonymousDelegate0`1[System.Nullable`1[System.Decimal]]
+                <>f__AnonymousDelegate0`1[System.Nullable`1[System.Decimal]]
+                System.Action`1[System.Nullable`1[System.Decimal]]
+                """).VerifyDiagnostics();
+            verifier.VerifyTypeIL("<>f__AnonymousDelegate0`1", s_anonymousDelegateWithDecimalConstant);
+        }
+
+        private static readonly string s_anonymousDelegateWithDateTimeConstant = $$"""
+            .class private auto ansi sealed '<>f__AnonymousDelegate0`1'<T1>
+                extends [{{s_libPrefix}}]System.MulticastDelegate
+            {
+                .custom instance void [{{s_libPrefix}}]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+                	01 00 00 00
+                )
+                // Methods
+                .method public hidebysig specialname rtspecialname 
+                	instance void .ctor (
+                		object 'object',
+                		native int 'method'
+                	) runtime managed 
+                {
+                } // end of method '<>f__AnonymousDelegate0`1'::.ctor
+                .method public hidebysig newslot virtual 
+                	instance void Invoke (
+                		[opt] !T1 arg
+                	) runtime managed 
+                {
+                	.param [1]
+                		.custom instance void [{{s_libPrefix}}]System.Runtime.CompilerServices.DateTimeConstantAttribute::.ctor(int64) = (
+                			01 00 64 00 00 00 00 00 00 00 00 00
```

Comment:
Same comment about what this constant represents.

Reply:
Should be apparent from the C# signature in the field's comment now.